### PR TITLE
restrict archiving form that creates cases if case has other forms

### PR DESCRIPTION
**Pull request by snopoke**
Prevent archiving a form if it creates any cases which have other forms updating them. To put it another way, you can only archive a form which creates a case if that form is the only form which touches the case.

The reason for this change is to avoid the scenario where a case exists but is missing the form that created it which results in some confusing (and incorrect) data.

### CODE REVIEW

**Review by benrudolph — APPROVED** [review]
so this eliminates the ability to erase the initial changes to a case that a form makes. the reason we're doing this is that it's somewhat weird to have the second "update" form create the case. this was an edge case during your migrations right? seems fine to me, wonder if we'll get any complaints

**Comment by czue** [comment]
I agree this makes sense and that it's better to ask for forgiveness than permission if it comes up (and the answer likely being "sorry, this was a bug that you could ever do it which we've fixed"). Unless someone makes a very compelling case I can't see us losing that debate.

Probably good to let flos know about the change in case anyone gets confused or reports a bug.

File: corehq/apps/reports/views.py [hunk 1]
```diff
@@ -1859,26 +1868,60 @@ def archive_form(request, domain, instance_id):
...
+
+def _get_cases_with_forms_message(domain, cases_with_other_forms, case_id_from_request):
+    def _get_case_link(case_id, name):
+        if case_id == case_id_from_request:
+            return _(u"%(case_name)s (this case)") % {'case_name': name}
+        else:
+            return u'<a href="{}#!history">{}</a>'.format(reverse('case_details', args=[domain, case_id]), name)
+
+    case_links = ', '.join([
+       _get_case_link(case_id, name)
```
Comment (dimagimon) [inline corehq/apps/reports/views.py]: E121 continuation line under-indented for hanging indent

**Comment by snopoke** [comment]
:white_check_mark:

**Pull request merged**